src/scripts/components/LanguagePanel/LinkTranslations.jsx

```diff
@@ -121,140 +121,149 @@ const SearchBar = ({ search, setSearch, refetch, disabled }) => {
 	);
 };
 
-const LinkTranslations = ({ postId, refetchLangs }) => {
-	const [isModalOpen, setIsModalOpen] = useState(false);
-	const { data, refetch, isLoading, isError } = useLinkablePosts(postId, 1);
-	const [page, setPage] = useState(1);
-	const [totalPages, setTotalPages] = useState(data?.pages || 1);
-	const [search, setSearch] = useState('');
+const LinkTranslations = ( { postId, refetchLangs } ) => {
+	const [ isModalOpen, setIsModalOpen ] = useState( false );
+	const { data, refetch, isLoading, isError } = useLinkablePosts( postId, 1, false );
+	const [ page, setPage ] = useState( 1 );
+	const [ totalPages, setTotalPages ] = useState( data?.pages || 1 );
+	const [ search, setSearch ] = useState( '' );
+
+	const openModal = () => {
+		if ( isLoading ) {
+			refetch( 1 );
```

Comment:
This is simpler if the logic is made in react redering or not a compoent, instead of using conditions and fetching blockers. Just make the Modal in another component that uses the hook that fetches stuff. Then with the existing logic, that hook only runs when the other component is rendered.

```jsx
const PostsModal = () => {
	const { data, refetch, isLoading, isError } = useLinkablePosts( postId, 1 );

	// Handle loading like any other page.

	return (
		<Modal ...>
	)
}

const LinkTranslations = () => {
	// All logic...

	return (
		<>
			{isModalOpen && <PostsModal/>}
		</>
	)
}
```